File: analytics/src/test/java/io/appmetrica/analytics/impl/utils/validation/NonEmptyStringValidatorTest.java
package io.appmetrica.analytics.impl.utils.validation;

import io.appmetrica.analytics.testutils.CommonTest;
import org.junit.Test;

import static org.assertj.core.api.Assertions.assertThat;

public class NonEmptyStringValidatorTest extends CommonTest {

    private final Validator<String> mValidator = new NonEmptyStringValidator("someTestObject");

    @Test
    public void testNull() {
        assertThat(mValidator.validate(null).isValid()).isFalse();
    }

    @Test
    public void testEmpty() {
        assertThat(mValidator.validate("").isValid()).isFalse();
    }

    @Test
    public void testValid() {
        assertThat(mValidator.validate("string").isValid()).isTrue();
    }

}

File: core-utils/src/testDebug/java/io/appmetrica/analytics/coreutils/internal/validation/NonEmptyStringValidatorTest.kt
package io.appmetrica.analytics.coreutils.internal.validation

import io.appmetrica.gradle.testutils.CommonTest
import org.assertj.core.api.Assertions.assertThat
import org.junit.Test

class NonEmptyStringValidatorTest : CommonTest() {

    private val validator = NonEmptyStringValidator("someTestObject")

    @Test
    fun nullAndEmptyFail() {
        assertThat(validator.validate(null).isValid).isFalse
        assertThat(validator.validate("").isValid).isFalse
    }

    @Test
    fun nonEmptyPasses() {
        assertThat(validator.validate("string").isValid).isTrue
    }
}
